packages/frontend-plugin-api/src/wiring/createPlugin.ts

```diff
@@ -78,14 +100,17 @@ export function createPlugin<
     $$type: '@backstage/BackstagePlugin',
     version: 'v1',
     id: options.id,
-    routes: options.routes ?? ({} as Routes),
-    externalRoutes: options.externalRoutes ?? ({} as ExternalRoutes),
+    routes: options.routes ?? ({} as TRoutes),
+    externalRoutes: options.externalRoutes ?? ({} as TExternalRoutes),
     featureFlags: options.featureFlags ?? [],
     extensions,
+    getExtension(id) {
```

Comment:
Do you think adding a method that lists the extension IDs would be useful? I was wondering if there is a way for people to know the extension IDs of the plugin. They call the method and get a list of IDs + a short description about the extension (similar to a help command output)...

Reply:
Idea is to have this available as metadata for the package. I'm thinking we should avoid it in the runtime API for now, but could be something to consider in the future if needed. My thought here being that you'll need to know exactly which extension you're overriding in code anyway, but it could be that someone comes up with some form of output decorator pattern that could require this.